fed/mosaic-output/src/main/resources/web/visualizer-dev.js

```diff
@@ -11,279 +11,304 @@ import {Icon, Style} from 'ol/style'
+
+        // Update style
+        const style = this.createStyle()
+        this.marker.setStyle(style)
+
+        if ((Date.now() - this.timeStateChange) > 500) {
+            // Clear sending/receiving states
+            this.state.sending = false
+            this.state.receiving =  false
+        }
 
-  },
+    },
 
-  /**
-   * Creates the style of the RSU marker based on the RSUs state.
-   */
-  createStyle() {
-    let style = 'roadside-unit'
-    if (this.state.equipped) {
-      style = 'roadside-unit-equipped'
-    }
-    if (this.state.sending) {
-      style = 'roadside-unit-sending'
-    }
-    if (this.state.receiving) {
-      style = 'roadside-unit-receiving'
+    /**
+     * Creates the style of the RSU marker based on the RSUs state.
+     */
+    createStyle() {
+        let style = 'roadside-unit'
+        if (this.state.equipped) {
+            style = 'roadside-unit-equipped'
+        }
+        if (this.state.sending) {
+            style = 'roadside-unit-sending'
+        }
+        if (this.state.receiving) {
+            style = 'roadside-unit-receiving'
+        }
+        return new Style({
+            image: new Icon({
+                anchor: [0.5, 1],
+                src: `markers/${ style }.png`
+            })
+        })
     }
-    return new Style({
-      image: new Icon({
-        anchor: [0.5, 1],
-        src: `markers/${ style }.png`
-      })
-    })
-  }
 }
 
 /**
  * Represents a vehicle.
  */
 const Vehicle = {
-  name: 'unnamed_vehicle',
-  latitude: 0,
-  longitude: 0,
-  marker: null,
-  timeStateChange: 0,
-  state: {},
-
-  init(name, latitude, longitude) {
-    this.name = name
-    this.latitude = latitude
-    this.longitude = longitude
-    this.marker = new Feature({
-      type: 'vehicle',
-      geometry: new Point(fromLonLat([ longitude, latitude ]))
-    })
-    this.marker.setProperties(['name', 'unit'])
-    this.marker.set('name', name)
-    this.marker.set('unit', this)
-    this.state = {
-      equipped: false,
-      sending: false,
-      receiving: false,
-      parking: false,
-      charging: false,
-    };
-  },
-
-  getMarker() {
-    return this.marker
-  },
-
-  setIsEquipped(isEquipped) {
-    this.state.equipped = isEquipped
-  },
-
-  setLocation(latitude, longitude) {
-    this.latitude = latitude
-    this.longitude = longitude
-  },
-
-  setState(stateName) {
-    if (this.state[stateName] !== undefined) {
-      this.state[stateName] = true
-      this.timeStateChange = Date.now();
-    }
-  },
-
-  /**
-   * Updates the marker of the vehicle.
-   */
-  updateView() {
-    // Location
-    this.marker.setGeometry(new Point(fromLonLat([ this.longitude, this.latitude ])))
-
-    // Update style
-    const style = this.createStyle()
-    this.marker.setStyle(style)
-
-    if ((Date.now() - this.timeStateChange) > 500) {
-      // Clear sending/receiving states
-      this.state.sending = false
-      this.state.receiving =  false
-      this.timeStateChange = 0
```

Comment:
Why did you remove this?

Reply:
The line has no effect, it will again run into this if block after the change. also mentioned this briefly in the commit msg